### PR TITLE
PILOT-1975: create folder component

### PR DESCRIPTION
**Pull request by hugoantunes**
## Summary

- dataset/schemas/folder.py -> dataset/components/folder/schemas.py
- dataset/routers/v1/api_dataset_folder.py -> dataset/components/folder/views.py
- add folder.schemas
  - FolderCreateSchema
  - FolderResponseSchema
  - LegacyFolderResponseSchema
  - MetadataCreateSchema
- add components.folder.activitiy_log.FolderActivityLog
- add components.folder.manager.FolderManager 
- add components.folder.dependencies.get_folder_manager

## JIRA Issues

https://indocconsortium.atlassian.net/browse/PILOT-1975

## Type of Change

Please delete options that are not relevant.

- [x] Refactor or reformatting

## Testing

Are there any new or updated tests to validate the changes?

- [ ] Yes
- [x] No

## Test Directions

- Folder creation should not be affected.

### CODE REVIEW

File: dataset/components/folder/activity_log.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    level_stderr=settings.LOG_LEVEL_STDERR,
+).get_logger()
+
+
+class FolderActivityLog(ActivityLogService):
+
+    topic = 'metadata.items.activity'
+    avro_schema_path = 'dataset/schemas/metadata.items.activity.avsc'
+
+    async def send_create_folder_event(self, folder: FolderResponseSchema, user: str):
```
Comment (afmam) [inline dataset/components/folder/activity_log.py:39]: Missing return type hint

File: dataset/components/folder/views.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    tags=['V1 DATASET'],
+    response_model=LegacyFolderResponseSchema,
+    summary='Create an empty folder',
+)
+async def create_folder(
+    dataset_id: str,
+    data: FolderCreateSchema,
+    dataset_crud: DatasetCRUD = Depends(get_dataset_crud),
+    folder_manager: FolderManager = Depends(get_folder_manager),
+):
```
Comment (afmam) [inline dataset/components/folder/views.py:51]: Missing return type hint

File: dataset/components/folder/activity_log.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    level_default=settings.LOG_LEVEL_DEFAULT,
+    level_file=settings.LOG_LEVEL_FILE,
+    level_stdout=settings.LOG_LEVEL_STDOUT,
+    level_stderr=settings.LOG_LEVEL_STDERR,
+).get_logger()
+
+
+class FolderActivityLog(ActivityLogService):
+
+    topic = 'metadata.items.activity'
```
Comment (afmam) [inline dataset/components/folder/activity_log.py:36]: Can we add this topic name to `config.py`?

File: dataset/services/metadata.py [hunk 4]
```diff
@@ -63,3 +76,10 @@ async def delete_object(cls, id_: str) -> None:
         async with httpx.AsyncClient() as client:
             response = await client.delete(url=cls.ITEM_URL, params={'id': id_})
         response.raise_for_status()
+
+    @classmethod
+    async def is_duplicated_name_item(cls, code: str, name: str, parent_id: str) -> bool:
```
Comment (afmam) [inline dataset/services/metadata.py:81]: Missing docstring

File: dataset/components/folder/activity_log.py (hunk 3 above)
Comment (hugoantunes) [reply dataset/components/folder/activity_log.py:36]: There is another ticket to rework the activity log, I could do this there.

File: dataset/services/metadata.py [hunk 5]
```diff
@@ -45,7 +54,11 @@ async def get_objects(cls, code: str, items_type: str = 'dataset') -> Dict[str,
     @classmethod
     async def get_by_id(cls, id_: str) -> Dict[str, Any]:
         url = f'{cls.ITEM_URL}{id_}/'
-        return (await cls.get(url))['result']
+        try:
+            return (await cls.get(url))['result']
+        except Exception:
+            logger.exception('Item not found')
+            raise NotFound()
```
Comment (afmam) [inline dataset/services/metadata.py:61]: The `raise_for_status` could potentially raise a 5xx error, which would raise a `NotFound` here. The metadata service would return 404 if the item doesn't exist. Is there a way to catch 404 specifically here?

**Review by afmam — CHANGES_REQUESTED** [review]
Left a few comments

File: dataset/components/folder/activity_log.py (hunk 3 above)
Comment (afmam) [reply dataset/components/folder/activity_log.py:36]: @hugoantunes sounds good

File: dataset/services/metadata.py (hunk 5 above)
Comment (hugoantunes) [reply dataset/services/metadata.py]: nice catch!

**Review by afmam — APPROVED** [review]
LGTM

**Pull request merged**